### DISCUSSION TITLE
v5.48.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#13913] [General]
# Features

* enable import assertions again

# Bugfixes

* upgrade webpack-sources for fixes regarding source maps
* fix infinite loop in HMR runtime code

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.48.0'>v5.48.0</a>.</em>

**Comment by mwmcode** [comment]
Hi @sokra 

HMR stops working when I upgrade to this version (5.48). (_downgrading to 5.47.1 works_).

it recompiles but doesn't swap/refresh

https://user-images.githubusercontent.com/13101565/128291424-7f7d1d0f-5eab-4652-87a7-97618baeed43.mov



```js
rules: [
        {
          test: /\.tsx?$/,
          use: [
            {
              loader: 'ts-loader',
              options: {
                transpileOnly: true,
              },
            },
          ],
          exclude: /node_modules/,
        },

// ....

plugins: [
      ...(env.test
        ? []
        : [
            new webpack.HotModuleReplacementPlugin(),
            new ReactRefreshWebpackPlugin(),
          ]),
    ],

// ...

devServer: {
      hot: true,
      hotOnly: true,

```
